lib/vm/src/instance/mod.rs

```diff
@@ -1126,114 +1030,50 @@ impl InstanceHandle {
-                VMMemory {
-                    from,
-                    instance_ref: Some(WeakOrStrongInstanceRef::Strong(instance)),
-                }
-                .into()
-            }
-            ExportIndex::Global(index) => {
-                let from = {
-                    if let Some(def_index) = instance_ref.module.local_global_index(*index) {
-                        instance_ref.globals[def_index].clone()
-                    } else {
-                        let import = instance_ref.imported_global(*index);
-                        import.from.clone()
-                    }
-                };
-                VMGlobal {
-                    from,
-                    instance_ref: Some(WeakOrStrongInstanceRef::Strong(instance)),
-                }
-                .into()
-            }
-        }
-    }
-
-    /// Return an iterator over the exports of this instance.
-    ///
-    /// Specifically, it provides access to the key-value pairs, where the keys
-    /// are export names, and the values are export declarations which can be
-    /// resolved `lookup_by_declaration`.
-    pub fn exports(&self) -> indexmap::map::Iter<String, ExportIndex> {
-        self.module().exports.iter()
+    /// Lookup an exported function with the specified function index.
+    pub fn function_by_index(&self, idx: FunctionIndex) -> Option<VMFunction> {
+        let instance = self.instance.as_ref();
+        let imports = instance.artifact.import_counts().functions as usize;
+        let (address, signature, vmctx, trampoline) = if idx.index() < imports {
+            let import = instance.imported_function(idx);
+            (
+                *(import.body),
+                import.signature,
+                import.environment,
+                import.trampoline,
+            )
+        } else {
+            let index = LocalFunctionIndex::new(idx.index() - imports);
+            let func = instance.artifact.functions().get(index)?;
+            (
+                *(func.body),
+                func.signature,
+                VMFunctionEnvironment {
+                    vmctx: instance.vmctx_ptr(),
+                },
+                Some(func.trampoline),
+            )
+        };
+        Some(VMFunction {
+            // Any function received is already static at this point as:
+            // 1. All locally defined functions in the Wasm have a static signature.
+            // 2. All the imported functions are already static (because
+            //    they point to the trampolines rather than the dynamic addresses).
+            kind: VMFunctionKind::Static,
+            address,
+            signature,
+            vmctx,
+            call_trampoline: trampoline,
+            instance_ref: Some(WeakOrStrongInstanceRef::Strong(self.instance().clone())),
+        })
+    }
+
+    /// Lookup an exported function with the given name.
+    /// TODO(0-copy): make return type lifetime be dependent on lifetime of self...
```

Comment:
Can live without this. `VMFunction` contains a strong reference to this instance.